### PR TITLE
Replace `Vec::from_hex` with `hex!`

### PR DESCRIPTION
**Pull request by Kixunil**
This makes the code less noisy and is a preparation for changing it to `const`-based literal. Because of the preparation, places that used variables to store the hex string were changed to constants.

There are still some instances of `Vec::from_hex` left - where they won't be changeable to `const` and where `hex!` is unavailable (integration tests). These may be dealt with later.

See also #1189

Note that while the change appears big it's nearly entirely mechanical, so should be pretty easy to review. (But I don't feel it's `trivial`.)

### CODE REVIEW

**Review by sanket1729 — DISMISSED** [review]
ACK 22ab8d71a6b1d30794ac9502fd9449423e2ee7f5. Thanks for exporting the pub macro in tests. I have versions of this in many downstream repos.

**Comment by Kixunil** [comment]
@sanket1729 oh crap, that is a remnant of my failed experiment. It doesn't work the way I intended. I want to change it to `hex_lit` crate if @TheBlueMatt agrees. You could use that one too.

File: bitcoin/src/lib.rs [hunk 1]
```diff
@@ -83,7 +83,7 @@ extern crate actual_serde as serde;
 #[cfg(test)]
 #[macro_use]
 mod test_macros;
-mod internal_macros;
+pub(crate) mod internal_macros;
```
Comment (tcharding) [inline bitcoin/src/lib.rs:86]: I believe this change is meaningless, or am I missing something. All modules are in scope in the crate they are in, right? What use is a module that cannot be accessed by other code in that crate?

Comment (Kixunil) [reply bitcoin/src/lib.rs:86]: Oh I thought it was `pub(crate)` before. Inner modules have access to the items of outer modules.

**Review by tcharding — DISMISSED** [review]
ACK f797947f4dab8e3f81e67e22e338219b62fa5784

**Review by sanket1729 — APPROVED** [review]
ACK f797947f4dab8e3f81e67e22e338219b62fa5784

**Comment by apoelstra** [comment]
Needs rebase after #1387

**Review by apoelstra — APPROVED** [review]
ACK 089a1e452dc1a99338122903205ae4f64d48d824

**Review by tcharding — APPROVED** [review]
ACK 089a1e452dc1a99338122903205ae4f64d48d824

**Review by sanket1729 — APPROVED** [review]
code review ACk 089a1e452dc1a99338122903205ae4f64d48d824

**Pull request merged**